File: src/sw.php
<?php

namespace WpRefs\SW;
/*

usage:

use function WpRefs\SW\sw_fixes;

*/

function sw_fixes($text)
{
    // ---
    // find == Marejeleo == replace by == Marejeo ==
    $text = preg_replace('/(=+)\s*Marejeleo\s*(\1)/i', '\1 Marejeo \1', $text);
    // ---
    return $text;
}

File: python/src/sw.py
"""
Swahili fixes module

Usage:
    from src.sw import sw_fixes
"""

import re


def sw_fixes(text):
    """
    Apply Swahili-specific fixes
    
    Args:
        text: Text to process
        
    Returns:
        Modified text
    """
    # find == Marejeleo == replace by == Marejeo ==
    text = re.sub(r'(=+)\s*Marejeleo\s*(\1)', r'\1 Marejeo \1', text, flags=re.IGNORECASE)
    
    return text
